### DISCUSSION TITLE
Random Timeout from socket

### DISCUSSION THREAD

**Discussion by Edvaytech** [miguelgrinberg/Flask-SocketIO#1611] [Q&A]
In socket it give random timeout error

socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/wsgi.py", line 566, in handle_one_response
socket_1         |     result = self.application(self.environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/middleware.py", line 60, in __call__
socket_1         |     return self.engineio_app.handle_request(environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/socketio/server.py", line 534, in handle_request
socket_1         |     return self.eio.handle_request(environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/server.py", line 375, in handle_request
socket_1         |     environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/socket.py", line 107, in handle_get_request
socket_1         |     start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/socket.py", line 147, in _upgrade_websocket
socket_1         |     return ws(environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/async_drivers/eventlet.py", line 20, in __call__
socket_1         |     return super(WebSocketWSGI, self).__call__(environ, start_response)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/websocket.py", line 130, in __call__
socket_1         |     self.handler(ws)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/engineio/socket.py", line 172, in _websocket_handler
socket_1         |     pkt = ws.wait()
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/websocket.py", line 788, in wait
socket_1         |     for i in self.iterator:
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/websocket.py", line 643, in _iter_frames
socket_1         |     message = self._recv_frame(message=fragmented_message)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/websocket.py", line 669, in _recv_frame
socket_1         |     header = recv(2)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/websocket.py", line 578, in _get_bytes
socket_1         |     d = self.socket.recv(numbytes - len(data))
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/greenio/base.py", line 366, in recv
socket_1         |     return self._recv_loop(self.fd.recv, b'', bufsize, flags)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/greenio/base.py", line 360, in _recv_loop
socket_1         |     self._read_trampoline()
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/greenio/base.py", line 331, in _read_trampoline
socket_1         |     timeout_exc=socket_timeout('timed out'))
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/greenio/base.py", line 210, in _trampoline
socket_1         |     mark_as_closed=self._mark_as_closed)
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/hubs/__init__.py", line 159, in trampoline
socket_1         |     return hub.switch()
socket_1         |   File "/usr/local/lib/python3.7/site-packages/eventlet/hubs/hub.py", line 298, in switch
socket_1         |     return self.greenlet.switch()
socket_1         | socket.timeout: timed out
socket_1         |

sometime it gives this error as well

socket_1         | 75e9ebbfd456438db4b88ad4d56476d5 {'wsgi.errors': <_io.TextIOWrapper name='<stderr>' mode='w' encoding='UTF-8'>, 'wsgi.version': (1, 0), 'wsgi.multithread': True, 'wsgi.multiprocess': False, 'wsgi.run_once': False, 'wsgi.url_scheme': 'http', 'REQUEST_METHOD': 'GET', 'SCRIPT_NAME': '', 'RAW_PATH_INFO': '/socket.io/', 'PATH_INFO': '/socket.io/', 'QUERY_STRING': 'EIO=3&transport=polling&t=NdFsGjk', 'CONTENT_TYPE': 'text/plain', 'SERVER_PROTOCOL': 'HTTP/1.0', 'SERVER_NAME': '172.21.0.10', 'SERVER_PORT': '5001', 'REMOTE_ADDR': '172.21.0.12', 'REMOTE_PORT': '37114', 'GATEWAY_INTERFACE': 'CGI/1.1', 'headers_raw': (('Host', 'wsklaymarket.edvay.com'), ('User-Agent', 'Mozilla/5.0 (Linux; Android 10; ONEPLUS A6000) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.210 Mobile Safari/537.36'), ('Accept', '*/*'), ('Accept-Encoding', 'gzip, deflate, br'), ('Accept-Language', 'en-US,en;q=0.9,kn;q=0.8'), 'wsgi.input': <eventlet.wsgi.Input object at 0x7f69a6ca92d0>, 'eventlet.input': <eventlet.wsgi.Input object at 0x7f69a6ca92d0>, 'eventlet.posthooks': []}

we are doing monkey patching as well

```
import eventlet
from socketio import WSGIApp
from edvay.settings import SOCKET_PORT
from edvay.sio_events import sio

eventlet.monkey_patch()
app = WSGIApp(sio)
eventlet.wsgi.server(eventlet.listen(('', int(SOCKET_PORT))), app)
```

```
from edvay.sio_app import sio


@sio.event
def connect(sid, environ):
    user_socket_id = environ['HTTP_USER_SOCKET_ID']
    print(sid, environ)
    sio.enter_room(sid, user_socket_id)


@sio.event
def disconnect(sid):
    # user_socket_id = environ['HTTP_USER_SOCKET_ID']
    # sio.leave_room(sid, user_socket_id)
    pass
```

```
from socketio import RedisManager, Server
from edvay.settings import REDIS_SOCKET_URL

mgr = RedisManager(REDIS_SOCKET_URL)  # , write_only=True)
sio = Server(client_manager=mgr, cors_allowed_origins='*')


def _send(user_socket_id, data, event):
    return sio.emit(event, {'data': data}, room=user_socket_id)


def sio_notification(user_socket_id, message):
    return _send(user_socket_id, message, 'notification')


def sio_chat(user_socket_id, data):
    return _send(user_socket_id, data, 'chat')
```

the version are

eventlet==0.25.1
redis==3.3.11
python-socketio==4.3.1

**Comment by miguelgrinberg** [comment]
This is a low-level issue within eventlet, so I'm not really sure I can help. Are your clients in slow/unstable networks? Maybe that's the cause, but I really don't know for sure.
